proj/hps_accel/gateware/stream/fifo.py

```diff
@@ -0,0 +1,63 @@
+#!/bin/env python
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from nmigen import signed, unsigned, Cat, Signal, Module
+from nmigen.lib.fifo import SyncFIFOBuffered
+from nmigen.hdl.dsl import FSM
+from nmigen_cfu import SimpleElaboratable
+
+from .stream import PayloadDefinition, Endpoint
+
+
+class StreamFifo(SimpleElaboratable):
```

Comment:
Hmm we can add this to the list of stuff that should be upstreamed in nmigen streams I guess :-)

Reply:
Yup! The whole stream directory is candidate for being reusable somewhere.